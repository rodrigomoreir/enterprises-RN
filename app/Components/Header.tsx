import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import images from '../Themes/Images';
import { getStyles } from './Header.style'

interface Props {
  title: string
  goBack?: () => void
  goProfile: () => void
  image?: Object
}

const Header: React.FC<Props> = ({ title, goBack, image=images.profileUser, goProfile }) => {
  const styles = getStyles()

  return (    
    <View style={styles.container}>
      <View style={{width: 40}}>
        {goBack && (
          <TouchableOpacity style={styles.imageContainer} onPress={goBack}>
            <Image source={images.chevronLeft} style={styles.image}/>
          </TouchableOpacity>
        )}
      </View>
      <Text style={styles.title}>{title}</Text>
      <TouchableOpacity style={styles.profileContainer} onPress={goProfile}>
        <Image source={image} style={styles.imageProfileUser}/>
      </TouchableOpacity>

    </View>
  )
}

export default Header;
