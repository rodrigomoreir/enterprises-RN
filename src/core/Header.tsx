import React from 'react'
import images from '../assets'
import styled from 'styled-components/native'
import addOpacity from '../utils/addOpacity'
import { screenWidth } from '../utils'
import themeColors from '../theme/theme'

interface Props {
  title: string
  goBack?: () => void
  goProfile: () => void
  image?: Object
}

const Header: React.FC<Props> = ({
  title,
  goBack,
  image = images.profileUser,
  goProfile,
}) => {
  return (
    <StyledContainer>
      <StyledView>
        {goBack && (
          <StyledImageContainer onPress={goBack}>
            <StyledImage source={images.chevronLeft} />
          </StyledImageContainer>
        )}
      </StyledView>
      <StyledTitle>{title}</StyledTitle>
      <StyledProfileContainer onPress={goProfile}>
        <StyledImageProfile source={image} />
      </StyledProfileContainer>
    </StyledContainer>
  )
}
// border-color: ${({ theme }) => theme.colors.grey};
// tint-color: ${({ theme }) => theme.colors.lightBlue};
// color: ${({ theme }) => theme.colors.darkGrey};

const StyledContainer = styled.View`
  width: ${screenWidth}px;
  height: 60px;
  align-items: center;
  justify-content: space-between;
  padding-horizontal: 10px;
  border-bottom-width: 2px;
  border-color: ${({ theme }) => theme.colors.grey};
  flex-direction: row;
`
const StyledView = styled.View`
  width: 40px;
`
const StyledImageContainer = styled.TouchableOpacity`
  background-color: ${addOpacity(themeColors.colors.blue, 0.8)};
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border-radius: 20px;
  border-width: 1px;
  border-color: ${addOpacity(themeColors.colors.darkGrey, 0.6)};
`
const StyledImage = styled.Image`
  height: 20px;
  width: 20px;
  tint-color: ${({ theme }) => theme.colors.lightBlue};
`
const StyledTitle = styled.Text`
  font-size: 20px;
  line-height: 26.38px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
  text-align: center;
`
const StyledProfileContainer = styled.TouchableOpacity`
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border-radius: 20px;
  border-width: 1px;
  border-color: ${addOpacity(themeColors.colors.darkGrey, 0.6)};
`
const StyledImageProfile = styled.Image`
  max-height: 38px;
  max-width: 38px;
  border-radius: 20px;
  tint-color: ${addOpacity(themeColors.colors.blue, 0.8)};
`
export default Header
