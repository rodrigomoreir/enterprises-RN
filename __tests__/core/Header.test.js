import React from 'react'

// import renderer from 'react-test-renderer'
import { fireEvent } from '@testing-library/react-native'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../src/functions/testing'

import Header from '../../src/core/Header'

describe('Header', () => {
  test('matches snapshot', () => {
    const onPressMock = jest.fn()
    matchSnapshotWithTheme(<Header title={'TEST'} goProfile={onPressMock} />)
  })

  test('renders correctly', () => {
    const onPressMock = jest.fn()
    const { getByText } = renderWithTheme(
      <Header title={'TEST'} goProfile={onPressMock} />,
    )
    const header = getByText('TEST')
    expect(header).toBeTruthy()
  })

  test('render header with goBack', () => {
    const goBackMock = jest.fn()
    renderWithTheme(
      <Header
        goProfile={jest.fn()}
        goBack={goBackMock}
        isLoading
        title={'TEST'}
      />,
    )
  })
})
