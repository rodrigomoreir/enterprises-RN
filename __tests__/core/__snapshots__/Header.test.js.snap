// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#a9a9a9",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "width": 750,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "width": 40,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#696969",
          "fontSize": 20,
          "fontStyle": "normal",
          "fontWeight": "bold",
          "lineHeight": 26.38,
          "textAlign": "center",
        },
      ]
    }
  >
    TEST
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#69696999",
        "borderRadius": 20,
        "borderWidth": 1,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/profile.png",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 20,
            "maxHeight": 38,
            "maxWidth": 38,
            "tintColor": "#87cefaCC",
          },
        ]
      }
    />
  </View>
</View>
`;
